lib/src/predicate.dart

```diff
@@ -5,17 +5,19 @@ import 'package:flutter/foundation.dart';
 class FieldPredicate<T> {
   final String _operand;
   String? _value;
-  List<String>? _values;
+  List<String?>? _values;
+  bool _useValue;
 
-  FieldPredicate.equals(T value) : _operand = '==', _value = value.toString();
-  FieldPredicate.notEquals(T value) : _operand = '!=', _value = value.toString();
-  FieldPredicate.oneOf(List<T> values) : _operand = 'oneOf', _values = values.map((value) => value.toString()).toList();
-  FieldPredicate.notOneOf(List<T> values) : _operand = '!oneOf', _values = values.map((value) => value.toString()).toList();
+  FieldPredicate.equals(T value) : _operand = '==', _value = (value != null ? value.toString() : null), _useValue = true;
+  FieldPredicate.notEquals(T value) : _operand = '!=', _value = (value != null ? value.toString() : null), _useValue = true;
+  FieldPredicate.oneOf(List<T> values) : _operand = 'oneOf', _values = values.map((value) => (value != null ? value.toString() : null)).toList(), _useValue = false;
+  FieldPredicate.notOneOf(List<T> values) : _operand = '!oneOf', _values = values.map((value) => (value != null ? value.toString() : null)).toList(), _useValue = false;
```

Comment:
This can be simplified:

```suggestion
  FieldPredicate.equals(T value) : _operand = '==', _value = value?.toString(), _useValue = true;
  FieldPredicate.notEquals(T value) : _operand = '!=', _value = value?.toString(), _useValue = true;
  FieldPredicate.oneOf(List<T> values) : _operand = 'oneOf', _values = values.map((value) => value?.toString()).toList(), _useValue = false;
  FieldPredicate.notOneOf(List<T> values) : _operand = '!oneOf', _values = values.map((value) => value?.toString()).toList(), _useValue = false;
```